### PR TITLE
JavaScript and style for right rail

### PR DESCRIPTION
**Pull request by tarvos21**
- JavaScript from docs.js
- Style modifications
- Dependency
- typo

### CODE REVIEW

File: content/static/stylesheets/default.sass [hunk 1]
```diff
@@ -130,6 +130,23 @@ img
...
+  left: 827px!important
+  .header
+    margin-top: 3em!important
+
+.ui.demo.container
+  margin-left: calc((100% - 1127px)/2) !important
+  margin-right: 387px!important
+  .tab
+    width: 827px
+    .example:last-child:not(.position)
```
Comment (xiaohanyu) [inline content/static/stylesheets/default.sass:147]: This line seems too large.

Normal page:

image

Compared with http://semantic-ui.com/kitchen-sink.html#/modules

image

Comment (tarvos21) [reply content/static/stylesheets/default.sass:147]: Yes, I set the padding-bottom of last example to 35em, to keep the "transition" item in right rail become "active" and change color. To make it more comfortable, I may apply some more specific definitions on the last example.

File: content/themes/semantic-ui/amazon/index.slim [hunk 2]
```diff
@@ -9,8 +9,9 @@
       a.item data-tab="views" Views
       a.item data-tab="modules" Modules
 
-.ui.demo.container
+.ui.demo.container.basic.segment
```
Comment (xiaohanyu) [inline content/themes/semantic-ui/amazon/index.slim:12]: Could you explain why would you like to make it a `.container.basic.segment`?

I checked http://semantic-ui.com/kitchen-sink.html page and found it is a `.basic.container`,

File: content/static/stylesheets/default.sass [hunk 3]
```diff
@@ -130,6 +130,23 @@ img
...
+      padding-bottom: 35em
+  &.basic.segment
```
Comment (xiaohanyu) [inline content/static/stylesheets/default.sass:149]: Found `.basic.segment` here, 😄

File: content/themes/semantic-ui/amazon/index.slim (hunk 2 above)
Comment (tarvos21) [reply content/themes/semantic-ui/amazon/index.slim:12]: No, I can't definitely explain it now. If not added, the right rail will go to the top of the page.

File: content/static/stylesheets/default.sass (hunk 1 above)
Comment (xiaohanyu) [reply content/static/stylesheets/default.sass:147]: OK, but I still don't quite understand the connections between `35em` and "transition" item in right rail ...

I think `35em` is too large and unacceptable. Do me a favor and figure out a better solution. Many thanks.

File: content/static/stylesheets/default.sass (hunk 3 above)
Comment (tarvos21) [reply content/static/stylesheets/default.sass:149]: Yes, making it a segment seems to make it easier to define the size of anything, at least for this page.

**Comment by xiaohanyu** [comment]
Found a typo in your commit message: https://github.com/semantic-ui-forest/semantic-ui-forest/pull/2/commits/05aca509df8cacdc3b410dea8e4b110e39e0cef2

`screnn wider` -> `screen wider`

File: content/static/stylesheets/default.sass (hunk 1 above)
Comment (tarvos21) [reply content/static/stylesheets/default.sass:147]: OK, I will try my best.

File: content/static/stylesheets/default.sass [hunk 4]
```diff
@@ -130,6 +130,23 @@ img
       .ui.header a
         color: inherit
 
+/* Position, width, margin modifications for right rail
+/* Only for screen wider than 1200px
```
Comment (xiaohanyu) [inline content/static/stylesheets/default.sass]: I'm not quite sure whether I asked the right question. You write "Only for screen wider than 1200px". But if the code only works for "screen wider than 1200px", should we wrap it in a CSS media query?

Ref:
https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries

File: content/static/javascripts/docs.js [hunk 5]
```diff
@@ -0,0 +1,451 @@
...
+    $sectionHeaders      = $container.children('h2'),
+    $followMenu          = $container.find('.following.menu'),
+    $sectionExample      = $container.find('.example'),
+    $exampleHeaders      = $sectionExample.children('h4'),
+    $footer              = $('.page > .footer'),
+    $pageTabs            = $('.vertical.segment .tabs.menu .item'),
+    $example             = $('.example'),
+
+   // alias
+    handler
```
Comment (xiaohanyu) [inline content/static/javascripts/docs.js:32]: lacks one space alignment for `// alias`

**Comment by xiaohanyu** [comment]
Oh, the js code is really long, need some time to review this carefully.

File: content/themes/semantic-ui/amazon/index.slim (hunk 2 above)
Comment (xiaohanyu) [reply content/themes/semantic-ui/amazon/index.slim:12]: It's OK, I think maybe you should keep this question in your head and try to figure it out in your spare time.

File: content/static/javascripts/docs.js [hunk 6]
```diff
@@ -0,0 +1,451 @@
...
+      },
+      text: function($element, text) {
+        $.each(text, function(selector, text) {
+          $element.find(selector).text(text);
+        });
+        return $element;
+      }
+    },
+
+      refreshSticky: function() {
```
Comment (xiaohanyu) [inline content/static/javascripts/docs.js:404]: Indentation error here.

File: content/static/javascripts/docs.js [hunk 7]
```diff
@@ -0,0 +1,451 @@
...
+            .removeClass('active')
+          ;
+          $activeSection
+            .addClass('active')
+          ;
+        }
+      }
+    },
+
+   tryCreateMenu: function(event) {
```
Comment (xiaohanyu) [inline content/static/javascripts/docs.js]: Indentation error here.

**Review by xiaohanyu — APPROVED** [review]

**Comment by xiaohanyu** [comment]
Wonderful job, thanks!

**Pull request merged**